commit fdc4aa9c298004ccd397fd2b451065d52cbf1ee0
Author: Alex Levkovich <alevkovich@tut.by>
Date:   Sun Sep 1 02:23:11 2013 -0400

    Fix shutdown on critical battery level
    
    powerdevil kcontrol and daemon doesn't use consistent value for
    action, make them both use same enumeration is more desirable than
    plain number.
    
    BUG: 307288
    FIXED-IN: 4.11.2
    Reviewed By Weng Xuetian

diff --git a/powerdevil/daemon/powerdevilcore.cpp b/powerdevil/daemon/powerdevilcore.cpp
index 7e2f517..08cc827 100644
--- a/powerdevil/daemon/powerdevilcore.cpp
+++ b/powerdevil/daemon/powerdevilcore.cpp
@@ -28,6 +28,8 @@
 #include "powerdevilpolicyagent.h"
 #include "powerdevilprofilegenerator.h"
 
+#include "daemon/actions/bundled/suspendsession.h"
+
 #include <Solid/Battery>
 #include <Solid/Device>
 #include <Solid/DeviceNotifier>
@@ -505,17 +507,17 @@ bool Core::emitBatteryChargePercentNotification(int currentPercent, int previous
     if (currentPercent <= PowerDevilSettings::batteryCriticalLevel() &&
         previousPercent > PowerDevilSettings::batteryCriticalLevel()) {
         switch (PowerDevilSettings::batteryCriticalAction()) {
-        case 3:
+        case PowerDevil::BundledActions::SuspendSession::ShutdownMode:
             emitRichNotification("criticalbattery", i18n("Battery Critical (%1% Remaining)", currentPercent),
                              i18n("Your battery level is critical, the computer will be halted in 30 seconds."));
             m_criticalBatteryTimer->start();
             break;
-        case 2:
+        case PowerDevil::BundledActions::SuspendSession::ToDiskMode:
             emitRichNotification("criticalbattery", i18n("Battery Critical (%1% Remaining)", currentPercent),
                              i18n("Your battery level is critical, the computer will be hibernated in 30 seconds."));
             m_criticalBatteryTimer->start();
             break;
-        case 1:
+        case PowerDevil::BundledActions::SuspendSession::ToRamMode:
             emitRichNotification("criticalbattery", i18n("Battery Critical (%1% Remaining)", currentPercent),
                              i18n("Your battery level is critical, the computer will be suspended in 30 seconds."));
             m_criticalBatteryTimer->start();
